src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubProducer.cs

```diff
@@ -119,6 +119,8 @@ public async Task Send(ProducerContext context)
                     DestinationAddress = _context.EndpointAddress
                 };
 
+                sendContext.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
This needs to be right before the StartSendActivity, since their pipes could delay sending and ultimately cause it be canceled if they time out, etc.